### ISSUE TITLE
Storysource renders (and rehighlight source) each time the story renders.

### ISSUE THREAD

**Issue by paradoxxxzero** [storybookjs/storybook#10740]
So I have this quite simple story :
```jsx

export const Story = () => {
  const [text, setText] = useState('')
  return (
    <input value={text} onChange={({ target: { value } }) => setText(value)} />
  )
}
```

and have noticed that storysource with default configuration renders each time I press a key in the field. (In fact each time the story renders).
I realized it because in this story long pressing a key will result in render freeze, due to spending quite some time in highlighting the same source over and over.
NB: I don't even have the addon panel open.

Here's the profile when holding a key: 
![image](https://user-images.githubusercontent.com/271144/81668133-7cc20b00-9444-11ea-9983-4fdbdd383d6d.png)

and the associated stack trace:
```
highlight (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:173060)
highlight (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:75942)
getCodeTree (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:41389)
SyntaxHighlighter (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:41465)
renderWithHooks (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:18989)
updateFunctionComponent (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:21220)
beginWork (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:22796)
beginWork$1 (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:27365)
performUnitOfWork (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:26340)
workLoopSync (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:26316)
performSyncWorkOnRoot (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:25942)
(anonymous) (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:15275)
unstable_runWithPriority (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:44981)
runWithPriority$1 (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:15225)
flushSyncCallbackQueueImpl (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:15270)
flushSyncCallbackQueue (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:15258)
scheduleUpdateOnFiber (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:25385)
enqueueSetState (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:16825)
push../node_modules/react/cjs/react.development.js.Component.setState (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:42514)
StoryPanel._this.listener (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:71655)
(anonymous) (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:78037)
handleEvent (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:78036)
(anonymous) (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:77923)
handler (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:70964)
handleEvent (vendors~main.ca2cc0486bf4dafa1ca7.bundle.js:71060)
postMessage (async)
send (index.js:160)
handler (index.js:125)
emit (index.js:135)
sendEvent (preview.js:33)
decorated (preview.js:68)
(anonymous) (hooks.js:241)
(anonymous) (client_api.js:136)
withSubscriptionTracking (client_api.js:164)
(anonymous) (hooks.js:241)
(anonymous) (client_api.js:135)
(anonymous) (hooks.js:269)
storyFn (story_store.js:396)
renderWithHooks (react-dom.development.js:14803)
updateFunctionComponent (react-dom.development.js:17034)
beginWork (react-dom.development.js:18610)
beginWork$1 (react-dom.development.js:23179)
performUnitOfWork (react-dom.development.js:22154)
workLoopSync (react-dom.development.js:22130)
performSyncWorkOnRoot (react-dom.development.js:21756)
(anonymous) (react-dom.development.js:11089)
unstable_runWithPriority (scheduler.development.js:653)
runWithPriority$1 (react-dom.development.js:11039)
flushSyncCallbackQueueImpl (react-dom.development.js:11084)
flushSyncCallbackQueue (react-dom.development.js:11072)
flushPendingDiscreteUpdates (react-dom.development.js:21847)
flushDiscreteUpdates (react-dom.development.js:21827)
finishEventHandler (react-dom.development.js:764)
batchedEventUpdates (react-dom.development.js:798)
dispatchEventForLegacyPluginEventSystem (react-dom.development.js:3568)
attemptToDispatchEvent (react-dom.development.js:4267)
dispatchEvent (react-dom.development.js:4189)
unstable_runWithPriority (scheduler.development.js:653)
runWithPriority$1 (react-dom.development.js:11039)
discreteUpdates$1 (react-dom.development.js:21887)
discreteUpdates (react-dom.development.js:806)
dispatchDiscreteEvent (react-dom.development.js:4168)
```
Versions:
```json
{
    "@storybook/addon-actions": "^5.3.18",
    "@storybook/addon-links": "^5.3.18",
    "@storybook/addon-storysource": "^5.3.18",
    "@storybook/addons": "^5.3.18",
    "@storybook/react": "^5.3.18"
}
```

**Comment by paradoxxxzero** [comment]
_To me this is not a question but a plain old bug..._

Anyway, this seems to be fixed in 6.0.0-beta.5.

**Comment by shilman** [comment]
> _To me this is not a question but a plain old bug..._

Probably, but I haven't confirmed

> Anyway, this seems to be fixed in 6.0.0-beta.5.

Ok, going to close this then
